### PR TITLE
Handle basic HTTP auth when using proxy

### LINKED ISSUE #189: Add a proxy with authentication option
It would be nice to allow Googler to query Google through an authenticated proxy.

At the moment, when I run something like `googler -p http://user:pwd@myproxy:port jarun` I get the following error:

```
[ERROR] Failed to connect to proxy server user:pwd@myproxy:port: [Errno -2] Name or service not known.
```

### PR DESCRIPTION
**Pull request by bobey**
Fixes #189

### CODE REVIEW

File: googler [hunk 1]
```diff
@@ -669,12 +670,27 @@ class GoogleConnection(object):
...
+
         if proxy:
             logger.debug('Connecting to proxy server %s', proxy)
-            self._conn = TLS1_2Connection(proxy, timeout=timeout)
+            self._conn = TLS1_2Connection(proxy_host_port, timeout=timeout)
 
             logger.debug('Tunnelling to host %s' % host_display)
-            self._conn.set_tunnel(host, port=port)
+
+            auth_header_value = 'Basic ' + base64.b64encode(proxy_user_pwd.encode('utf-8')).decode('utf-8')
```
Comment (jarun) [inline googler]: Please add this line below `if auth_needed:` check.

**Comment by jarun** [comment]
Also, please fix the issue reported by Travis.

**Comment by bobey** [comment]
@jarun done

**Comment by jarun** [comment]
Thank you!

**Pull request merged**